ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/CreateSourceCommand.java

```diff
@@ -163,15 +171,49 @@ static void checkMetaData(
     }
   }
 
-  private RegisterTopicCommand registerTopicFirst(
-      final CreateSourceProperties properties
-  ) {
+  private void checkTopicExists(final CreateSourceProperties properties) {
     final String kafkaTopicName = properties.getKafkaTopic();
     if (!kafkaTopicClient.isTopicExists(kafkaTopicName)) {
       throw new KsqlException("Kafka topic does not exist: " + kafkaTopicName);
     }
+  }
 
-    return new RegisterTopicCommand(this.topicName, false, properties);
+  protected boolean needsCreateTopic() {
+    // ksqlTopic is referenced by the property REGISTERED_TOPIC which uses a topic already
+    // registered
+    return !properties.getKsqlTopic().isPresent();
+  }
+
+  protected void createTopic(final MutableMetaStore metaStore) {
+    if (metaStore.getTopic(topicName) != null) {
```

Comment:
i'm wondering if the default should be the old behavior of `IF NOT EXISTS` (e.g. not failing if it already is registered)

Reply:
You meant this one, right?
```
    if (metaStore.getTopic(topicName) != null) {	
      // Check IF NOT EXIST is set, if set, do not create topic if one exists.	
      if (notExists) {	
        return new DdlCommandResult(true, "Topic already registered.");	
      } else {	
        final String sourceType = getSourceType(metaStore);	
        final String errorMessage =	
            String.format("%s with name '%s' already exists", sourceType, topicName);	
        throw new KsqlException(errorMessage);	
      }	
    }
```
The `notExists` flag was set to False by the `CreateSourceCommand`. The flag was used only by the REGISTER TOPIC command (passed through the AstBuilder#visitRegisterTopic()). I don't know what's the use of that flag without the command.

Btw, I run the commands you asked:
```
ksql> create stream stream1(id int) with (kafka_topic='t1', value_format='delimited');

 Message        
----------------
 Stream created 
----------------

ksql> create stream stream1(id int) with (kafka_topic='t1', value_format='delimited');
Cannot create stream 'STREAM1': A stream with name 'STREAM1' already exists

ksql> create table table1(id int) with (kafka_topic='t1', value_format='delimited');

 Message       
---------------
 Table created 
---------------

ksql> create table table1(id int) with (kafka_topic='t1', value_format='delimited');
Cannot create table 'TABLE1': A table with name 'TABLE1' already exists

ksql> show streams;

 Stream Name         | Kafka Topic                 | Format    
---------------------------------------------------------------
 STREAM1             | t1                          | DELIMITED 
 KSQL_PROCESSING_LOG | default_ksql_processing_log | JSON      
---------------------------------------------------------------
ksql> show tables;

 Table Name | Kafka Topic | Format    | Windowed 
-------------------------------------------------
 TABLE1     | t1          | DELIMITED | false    
-------------------------------------------------
```

And also with a topic that does not exist:
```
ksql> create stream stream2(id int) with (kafka_topic='t0', value_format='delimited');
Topic 't0' does not exist.

ksql> create table table2(id int) with (kafka_topic='t0', value_format='delimited');
Topic 't0' does not exist. ...
```